src/Helper/HtmlMarkupSerializationHelper.php

```diff
@@ -173,4 +180,73 @@ private static function arrayToHtml($object, $key, $spaces, $indent, $traversabl
         }
         return $markup;
     }
+
+    /**
+     * Checks if a class variable can be encoded.
+     * In practice checks that a getter and a setter for the property
+     * is found from the class.
+     *
+     * @param $class
+     * @param $variable
+     * @return bool
+     */
+    public static function encodableClassVariable($class, $variable)
+    {
+        $getter = 'get' . str_replace('_', '', $variable);
+        $setter = 'set' . str_replace('_', '', $variable);
+        if (!method_exists($class, $getter) || !method_exists($class, $setter)) {
+            return false;
+        }
+
+        return true;
+    }
+
+    /**
+     * @param array|string|StringCollection $val
+     * @return array|string|StringCollection
+     * @throws NostoException
+     */
+    public static function encodeHtmlEntities($val)
+    {
+        if ($val instanceof StringCollection) {
+            $encodedCollection = clone $val;
+            $encodedCollection->setData(null);
```

Comment:
The annotation of `setData` defines an array as argument, you may fix the annotation (The actual implementation accepts a null value) which is misleading.